### PR TITLE
Handling various errors

### PR DESCRIPTION
**Pull request by AdityaHegde**
Needs changes from #1335

- [x] Adding errors to measures and dimensions.
- [x] Fixing model sql not updating editor.
- [x] Adding reactivity from model changes to metrics.

### CODE REVIEW

File: web-local/src/lib/application-state-stores/metrics-internal-store.ts [hunk 1]
```diff
@@ -152,6 +170,58 @@ export class MetricsInternalRepresentation {
...
+          dimensionErroredIndices.add(index);
+          this.internalRepresentationDocument.setIn(
+            ["dimensions", index, "__ERROR__"],
+            error.message
+          );
+          break;
+      }
+    }
+
+    // remove previous errors if set
```
Comment (ericpgreen2) [inline web-local/src/lib/application-state-stores/metrics-internal-store.ts:197]: Wouldn't it be more clear and concise to _first_ remove all existing errors, _then_ set errors?

**Comment by ericpgreen2** [comment]
When I write a _valid_ model query, the result set doesn't update.

File: web-local/src/lib/application-state-stores/metrics-internal-store.ts (hunk 1 above)
Comment (AdityaHegde) [reply web-local/src/lib/application-state-stores/metrics-internal-store.ts:197]: You are right. An instance of this class is created and the yaml documented regenerated every time. So this is not needed at all.

**Comment by AdityaHegde** [comment]
This PR doesnt deal with invalidations like that @ericpgreen2 . Actually will just merge #1331 to this to avoid confusion

**Review by ericpgreen2 — APPROVED** [review]

**Review by ericpgreen2 — APPROVED** [review]

**Pull request merged**